### PR TITLE
Exclude files of nested rvc repos from repo_files() ##projects

### PR DESCRIPTION
**Pull request by RHL120**
**Checklist**

- [ ] Closing issues: #issue
- [X] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**
This PR prevents repo_files() from traversing dirs with a .rvc in them (besides the root)

### CODE REVIEW

File: libr/core/rvc.c [hunk 1]
```diff
@@ -355,46 +355,65 @@ static bool rm_empty_dir(const char *rp) {
 	return true;
 }
 
-static RList *repo_files(const char *rp) {
-	RList *files = r_file_lsrf (rp);
-	if (!files) {
-		return NULL;
+//This is why anonymous functions are cool
```
Comment (trufae) [inline libr/core/rvc.c:358]: This is not anonymous. Anonymous means it have no name. And C doesnt support such things. Thats just a static/local function

File: libr/core/rvc.c [hunk 2]
```diff
@@ -355,46 +355,65 @@ static bool rm_empty_dir(const char *rp) {
...
-
 	}
-	free (repo_meta);
 	r_list_free (files);
 	return ret;
+}
 
+static RList *repo_files(const char *dir) {
+	RList *ret = r_list_new ();
+	if (!ret) {
```
Comment (trufae) [inline libr/core/rvc.c:409]: Remove this

File: libr/core/rvc.c [hunk 3]
```diff
@@ -355,46 +355,65 @@ static bool rm_empty_dir(const char *rp) {
...
+		return NULL;
+	}
+	if (!dr (ret, dir)) {
```
Comment (trufae) [inline libr/core/rvc.c:412]: Use if (ret && !dr..

File: libr/core/rvc.c [hunk 4]
```diff
@@ -898,20 +917,13 @@ R_API bool r_vc_branch(const char *rp, const char *bname) {
...
-		eprintf ("Repo already exists at: %s", path);
-		return false;
-	case -1:
-		eprintf ("Can't create repo\n");
-		return false;
-	case -2:
-		eprintf ("Repository exists but is corrupt\n");
+	char *vcp = r_str_newf ("%s" R_SYS_DIR ".rvc", path);
+	if (r_file_is_directory (vcp)) {
+		eprintf ("A directory already exists in %s\n", path);
```
Comment (trufae) [inline libr/core/rvc.c:922]: A repository better than a directory

File: libr/core/rvc.c (hunk 1 above)
Comment (RHL120) [reply libr/core/rvc.c:358]: yes lol, that't the point I wish C had Anonymous functions becaus if it did I would have been able to embed dr into repo_files...

Comment (trufae) [reply libr/core/rvc.c:358]: Remove the comment and find a better name for the function

File: libr/core/rvc.c (hunk 2 above)
Comment (RHL120) [reply libr/core/rvc.c:409]: Remove repo_files? but if I did we would habe to alloc everytime we list the files. That would be 9 lines per repo_file.

File: libr/core/rvc.c (hunk 1 above)
Comment (RHL120) [reply libr/core/rvc.c:358]: What should that name be?

File: libr/core/rvc.c [hunk 5]
```diff
@@ -355,46 +355,61 @@ static bool rm_empty_dir(const char *rp) {
...
 			break;
 		}
-
 	}
-	free (repo_meta);
 	r_list_free (files);
 	return ret;
+}
 
+static RList *repo_files(const char *dir) {
```
Comment (radare) [inline libr/core/rvc.c:406]:
```c
static RList *repo_files(const char *dir) {
 	RList *ret = r_list_newf (free);
 	if (ret) {
           if ( !traverse_files (ret, dir)) {
 		r_list_free (ret);
 		ret = NULL;
           }
 	}
 	return ret;
 }
```

maybe like this?

**Pull request merged**